File: 2244. Minimum Rounds to Complete All Tasks/Solution2244.java
class Solution2244 {
    public int minimumRounds(int[] tasks) {
        Map<Integer, Integer> freq = new HashMap<>();
        for (int task : tasks) {
            freq.put(task, freq.getOrDefault(task, 0) + 1);
        }
        int res = 0;
        for (int k : freq.keySet()) {
            int count = freq.get(k);
            if (count == 1) {
                return -1;
            } else {
                res += (count + 2) / 3;
            }
        }
        return res;
    }
}
/*
freq = 3*m  -> m
       3m+1 -> (m-1) + 2 = m+1
       3m+2 -> m + 1
*/

File: 2244. Minimum Rounds to Complete All Tasks/Solution2244.go
package main

import (
	"fmt"
)

func main() {
	tasks := []int{2, 2, 3, 3, 2, 4, 4, 4, 4, 4}
	res := minimumRounds(tasks)
	fmt.Println(" input:", tasks)
	fmt.Println("output:", res)
}

func minimumRounds(tasks []int) int {
	res, count := 0, make(map[int]int)
	for _, t := range tasks {
		count[t]++
	}
	for _, v := range count {
		if v == 1 {
			return -1
		}
		res += (v + 2) / 3
	}
	return res
}
